plugins/inputs/wireless/wireless_mac.go

```diff
@@ -0,0 +1,78 @@
+//go:build darwin
+// +build darwin
+
+package wireless
+
+import (
+	"errors"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// default executable path
+const (
+	OSXCMD = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport -I"
+)
+
+func (w *Wireless) exe_cmd(cmd string, wg *sync.WaitGroup) ([]byte, error) {
+	parts := strings.Fields(cmd)
+	head := parts[0]
+	parts = parts[1:]
+	out, err := exec.Command(head, parts...).Output()
+	wg.Done() // Need to signal to waitgroup that this goroutine is done
+	return out, err
+}
+
+func (w *Wireless) Gather(acc telegraf.Accumulator) error {
+	if runtime.GOOS == "darwin" {
+		// collect MAC OS wireless data
+		wg := new(sync.WaitGroup)
+		wg.Add(3)
+		wireless, err := w.exe_cmd(OSXCMD, wg)
+		if err != nil {
+			return err
+		}
+		metrics, tags, err := w.loadMacWirelessTable(wireless, w.DumpZeros)
+		if err != nil {
+			return err
+		}
+		acc.AddGauge("wireless", metrics, tags)
+		return nil
+	}
+	return errors.New("OS Not Supported")
+}
+
+func (w *Wireless) loadMacWirelessTable(table []byte, dumpZeros bool) (map[string]interface{}, map[string]string, error) {
+	metrics := strings.Split(string(string(table)), "\n")
+	tags := make(map[string]string)
+	points := make(map[string]interface{})
+	for x := 0; x < len(metrics); x++ {
+		fm := strings.Split(strings.TrimSpace(metrics[x]), ":")
+		if len(fm) > 1 {
+			name := strings.Replace(strings.Trim(strings.TrimSpace(fm[0]), ":"), " ", "_", -1)
+			v := strings.TrimSpace(fm[1])
+			val, err := strconv.Atoi(v)
+			if err == nil { // it's a number
+				if !dumpZeros && val == 0 {
+					continue
+				}
+				points[name] = int64(val)
+			} else { // it's a string
+				tags[name] = strings.Replace(v, " ", "_", -1)
+			}
+		}
+	}
+	tags["interface"] = "airport"
```

Comment:
This doesn't add anything useful. If a user wants to add an additional tag, they could do with `inputs.wireless.tags` config.

Reply:
Linux interfaces are automatically added `en0`, `en1`, etc. but there is no `default` interface on macOS, so adding this _only_ to macOS allows for filtering when you have input from both macOS and linux.

Reply:
If you want to be compatible between the two systems, you should have all the same tags and fields. Which is not really possible in this case.

It is possible to filter on when tags don't exist on a metric though. So there is no real reason why this tag should be added by default.

Reply:
I'm not shooting for 100% compatible, since that is impossible as you point out. But having an interface tag seems a minimum to me.

Reply:
I will wait until you two decide this before checking in further changes in order to avoid polluting the PR further.

Reply:
Hmmm I see both points. However, as this is a static tag for a single telegraf instance and you can still add a static tag in your config (via the [tags option in your input plugin](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#input-plugins)) I tend to agree to @Hipska... So please either get that name from somewhere (if any) or remove it...

Reply:
Cool, removed. Checkin coming shortly with all these changes.